core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -104,13 +90,72 @@ object RequestChannel extends Logging {
       }
     }
 
-    trace(s"Processor $processor received request: ${requestDesc(true)}")
+    trace(s"Processor $processor received request: ${summary.getDescription(true)}")
+
+    val summary: RequestSummary = new RequestSummary(
```

Comment:
Instead of building this inside `Request`, we should consider having a `Request.apply` method that builds `RequestSummary` and passes it as a parameter to `Request`. That might make it easier to understand what is part of `Request` and what is part of `RequestSummary`.